flask_admin/theme.py

```diff
@@ -0,0 +1,20 @@
+import typing
+from dataclasses import dataclass
+from functools import partial
+
+
+@dataclass
+class Theme:
```

Comment:
BaseTheme ?

Reply:
I think just `Theme` is OK - it could actually be used directly rather than just as a base class. If we had a Tabler theme that needs no config, I think it would read better as `Admin(theme=Theme('tabler'))` rather than `Admin(theme=BaseTheme('tabler'))`?

Reply:
I thought even that Tabler theme should inherit Theme, and do it like this:
```python
Admin(theme=TablerTheme())
```

Reply:
We announce the new themes through the changelogs, so, when importing from the module `theme`, they are able to get all themes by knowing with the class names like (not Theme or BaseTheme at all):
```python
from flask_admin.theme import Bootstrap5Theme, TablerTheme, TailwindTheme, TailwindTheme, TablerWithBootstrap4Theme
```
This was the way I thought this theme feature would look like.
But the current implementation is also awesome.

Reply:
The disadvantage of this, of course but it requires more coding.

Reply:
Maybe considering both methods like `Theme('ThemeName', **params)` and `ThemeName(**params)`.
It gives us some complexity, but I haven't really analyzed how convenient this can be for the users.